[DOCU-2308] DecK Konnect guide

lena-larionova opened the pull request:
### Summary

Adding a guide for using decK with Konnect with examples into the decK docs.

The only thing I couldn't figure out was the purpose of the `--konnect-addr` flag in the current environment. Users only have one API they can target, right? Or can they somehow target `konnect.konghq.com` still? I couldn't get it to do anything different - @GGabriele, is there something I'm missing, or is this flag not helpful to customers at the moment (only useful internally)?

Note that this is going into live docs, not into the `konnect` branch, because this version of decK is already out - even though Konnect is in beta.

### Reason
https://konghq.atlassian.net/browse/DOCU-2308

### Testing
Tested all the things with decK 1.12 and cloud.konghq.com; that's also how I figured out precedence for auth methods. If that's not the desired precedence, then it's something that we should know about anyway. 

Preview: https://deploy-preview-4030--kongdocs.netlify.app/deck/latest/guides/konnect/

GGabriele commented:
> The only thing I couldn't figure out was the purpose of the --konnect-addr flag in the current environment. Users only have one API they can target, right? Or can they somehow target konnect.konghq.com still? I couldn't get it to do anything different 

Two reasons:
- target the dev environment (internal only)
- target konnect.konghq.com

GGabriele commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,229 @@
+Successfully Konnected as MyName (Konnect Org)!
+```
+
+## Runtime groups
````
May be worth adding an example using `--konnect-runtime-group-name` to dump a config from a non-default runtime group?

lena-larionova commented:
> Two reasons:
target the dev environment (internal only)
target konnect.konghq.com

We won't be documenting how to target the internal one, so that's fine. 
To target konnect.konghq.com, is it simply `--konnect-addr konnect.konghq.com`?

GGabriele commented:
> > Two reasons:
> > target the dev environment (internal only)
> > target konnect.konghq.com
> 
> We won't be documenting how to target the internal one, so that's fine. To target konnect.konghq.com, is it simply `--konnect-addr konnect.konghq.com`?

`--konnect-addr https://konnect.konghq.com`

cloudjumpercat commented on src/deck/guides/konnect.md:
```diff
@@ -0,0 +1,227 @@
+
+### Authenticate with CLI flags
+
+There are two ways to authenticate through CLI flags: by passing the password in `--konnect-password` directly, or by saving it to a file and passing the filename to decK with `--konnect-password-file`.
```
```suggestion
There are two ways to authenticate through CLI flags. You can pass the password in `--konnect-password` directly or save it to a file and pass the filename to decK with `--konnect-password-file`.
```
Just reworked this a bit to see if I could remove the comma between the dependent clauses (I think that's right, but not 100% sure).

cloudjumpercat commented on src/deck/guides/konnect.md:
```diff
@@ -0,0 +1,227 @@
+
+There are two ways to authenticate through CLI flags: by passing the password in `--konnect-password` directly, or by saving it to a file and passing the filename to decK with `--konnect-password-file`.
+
+With `konnect-password`:
```
```suggestion
The following example shows how to use `konnect-password` to authenticate:
```

lena-larionova replied:
This is a bit wordy, but I can see where you're going with it. I'll try to reword to add a bit more context to this and the next example.

cloudjumpercat commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,227 @@
+  --konnect-password YOUR_PASSWORD
+```
+
+Or, save your {{site.konnect_short_name}} password to a file, then pass the file using the `--konnect-password-file` flag:
````
```suggestion
The following example shows how to save your {{site.konnect_short_name}} password to a file, then pass the file using the `--konnect-password-file` flag:
```

cloudjumpercat commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,227 @@
+
+```sh
+deck ping \
+  --konnect-email YOUR_EMAIL \
````
Should this be "email@example.com" based on our discussion about placeholder examples?

cloudjumpercat commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,227 @@
+
+```sh
+deck ping \
+  --konnect-email YOUR_EMAIL \
````
Same comment here about if this should be "email@example.com" or something like that.

cloudjumpercat commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,227 @@
+konnect-password: YOUR_PASSWORD
+```
+
+Alternatively, you can save your password in a separate file, then specify the password file instead of a password:
````
```suggestion
Alternatively, you can save your password in a separate file, then specify the password file instead of a password. 
For example:
```
Just to mirror the one above.

cloudjumpercat commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,227 @@
+konnect-password-file: PATH/TO/FILENAME
+```
+
+decK automatically uses the credentials in `~/.deck.yaml` in any subsequent calls:
````
```suggestion
decK automatically uses the credentials in `~/.deck.yaml` in any subsequent calls.
For example:
```

cloudjumpercat commented on src/deck/guides/konnect.md:
```diff
@@ -0,0 +1,227 @@
+
+## Runtime groups
+
+Target runtime groups in your state file with the `konnect_runtime_group` parameter:
```
If this is an option you can add to your basic config file, I might add this as a bullet under step one with something like "If you want to target runtime groups....". I just wasn't sure how this related to the steps below.

lena-larionova replied:
It's more about targeting custom runtime groups. By default, a file will target the runtime group `default`, and if you want to change that, you have to set it either in the file or with a flag. Even if you don't explicitly set it though, decK will add the parameter.

cloudjumpercat commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,227 @@
+  runtime_group_name: staging
+```
+
+You can also set this parameter using the `--konnect-runtime-group-name` flag:
````
Same comment as the one above about if this could be added as an optional "you could also add this" bullet under step one.

cloudjumpercat commented on src/deck/guides/konnect.md:
```diff
@@ -0,0 +1,227 @@
+If you leave this empty, don't provide a flag, or don't include a `_konnect` section at all, decK targets the `default` runtime group.
+You can see this by pushing a config file to {{site.konnect_short_name}} with `deck sync`.
+
+1. Create a basic config file:
```
```suggestion
1. Create a basic config file. For example:
```
Just to match other task lists I've seen where the command ends in a period. Not sure if that's standard or not.

lena-larionova replied:
Ack - that's just inconsistency on my part.

cloudjumpercat commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,227 @@
+      host: mockbin.org
+    ```
+
+2. Sync the file to {{site.konnect_short_name}}:
````
```suggestion
2. Sync the file to {{site.konnect_short_name}}.
```

cloudjumpercat commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,227 @@
+    deck sync
+    ```
+
+3. Pull down the configuration with `deck dump`:
````
```suggestion
3. Pull down the configuration with `deck dump`.
```

cloudjumpercat commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,227 @@
+    deck dump --konnect-runtime-group-name default
+    ```
+
+    See the newly added `_konnect` section:
````
```suggestion
    See the newly added `_konnect` section. For example:
```
One question I had is why they need to see the new section. Do they need to use that info in another spot?

lena-larionova replied:
In hindsight, this entire section is irrelevant. It's literally just here to illustrate what happens if you don't provide a group, but there's no reason to document that with steps.

cloudjumpercat commented on src/deck/guides/konnect.md:
```diff
@@ -0,0 +1,227 @@
+
+decK manages Gateway services, which contain configurations for the Gateway proxy.
+
+Although decK doesn't _directly_ manage {{site.konnect_short_name}} services, you can use tags to associate a Gateway service to a {{site.konnect_short_name}} service:
```
```suggestion
Although decK doesn't _directly_ manage {{site.konnect_short_name}} services, you can use tags to associate a Gateway service to a {{site.konnect_short_name}} service. For example:
```

cloudjumpercat commented on src/deck/guides/konnect.md:
```diff
@@ -0,0 +1,227 @@
+
+### Workspace connection refused
+
+When migrating from {{site.base_gateway}} to {{site.konnect_short_name}}, make sure to remove any `_workspace`tags. If you leave `_workspace` in, you get the following error:
```
```suggestion
When migrating from {{site.base_gateway}} to {{site.konnect_short_name}}, make sure to remove any `_workspace` tags. If you leave `_workspace` in, you get the following error:
```

cloudjumpercat commented on src/deck/guides/konnect.md:
```diff
@@ -0,0 +1,227 @@
+
+You can now sync the file as-is to apply it to the default runtime group, or add a key to apply the configuration to a specific runtime group.
+
+To apply configuration to custom runtime groups, replace `_workspace` with `runtime_group_name: GroupName`.
```
```suggestion
To apply the configuration to custom runtime groups, replace `_workspace` with `runtime_group_name: GroupName`.
```

cloudjumpercat commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,227 @@
+_workspace: staging
+```
+
+Into:
````
```suggestion
To:
```
Could be a personal preference though, so feel free to ignore.

lena-larionova commented:
@cloudjumpercat Reworked a bunch of things, your feedback made me realize that some of the content was organized in a confusing way. 

I removed the steps with the `deck sync` and `deck dump` examples, as they weren't really adding any value - the point of this topic is to serve as a reference, and there's no reason to do those steps other than to validate decK's behavior. I also split out the two CLI flag authentication options; not really sure why I had them lumped together in the first place.

lena-larionova commented on src/deck/guides/konnect.md:
```diff
@@ -0,0 +1,199 @@
+These commands are deprecated and have been replaced with flags.
+
+`--konnect-addr`
+:  Address of the {{site.konnect_short_name}} endpoint. (Default: `"https://us.api.konghq.com"`)
```
^ That's what I'm going with for now. Keeping the doc as-is till we fix it in the source.

lena-larionova commented on src/deck/guides/konnect.md:
```diff
@@ -0,0 +1,199 @@
+decK looks for {{site.konnect_short_name}} credentials in the following order of precedence:
+
+1. Password set with the `--konnect-password` flag
+2. decK configuration file (default: `~/.deck.yaml`)
```
The latter -  decK looks for a .deck.yaml file in the home directory, it doesn't create one. Do you think it would be useful to add that clarification?

lena-larionova commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,199 @@
+```sh
+deck ping \
+  --konnect-email example@example.com \
+  --konnect-password-file /PATH/TO/FILE
````
It doesn't! I tested with a couple of different things (no extension, .txt, .yaml, .md) and they all worked.

haydenlam commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,201 @@
+
+```yaml
+tags:
+- _KonnectService:example
````
```suggestion
- _KonnectService:<Konnect-Service-Name>
```

haydenlam commented on src/deck/guides/konnect.md:
````diff
@@ -0,0 +1,201 @@
+tags:
+- _KonnectService:example
+```
+If the {{site.konnect_short_name}} service doesn't exist, setting this tag creates a {{site.konnect_short_name}} service.
````
```suggestion
The `<Konnect-Service-Name>` identifies which {{site.konnect_short_name}} to associate the Gateway service to. If the {{site.konnect_short_name}} service doesn't exist, setting this tag creates a {{site.konnect_short_name}} service.
```

haydenlam commented on src/deck/guides/konnect.md:
```diff
@@ -0,0 +1,201 @@
+
+In {{site.konnect_short_name}}, there are two types of services:
+* Gateway services: Managed through Runtime Manager
+* {{site.konnect_short_name}} services: Managed through Service Hub
```
```suggestion
* {{site.konnect_short_name}} services: Managed through Service Hub
Each Konnect Service may contain one or more service versions. A service version represents an implementation of a Gateway service.
```

haydenlam commented on src/deck/guides/konnect.md:
```diff
@@ -0,0 +1,201 @@
+
+decK manages Gateway services, which contain configurations for the Gateway proxy.
+
+Although decK doesn't _directly_ manage {{site.konnect_short_name}} services, you can use tags to associate a Gateway service to a {{site.konnect_short_name}} service:
```
```suggestion
Although decK doesn't _directly_ manage {{site.konnect_short_name}} services or service versions, you can use tags to associate a Gateway service to a service version in a {{site.konnect_short_name}} service:
```
